packages/suite/src/components/suite/modals/TransactionDetail/components/IODetails.tsx

```diff
@@ -18,191 +22,382 @@ export const blurFix = css`
-            </IOGridCell>
-            <IOGridCell isAccountOwned={isAccountOwned}>
+        <GridItem isAccountOwned={isAccountOwned}>
+            <IOAddress
+                txAddress={addresses?.length ? addresses[0] : ''}
+                explorerUrl={explorerTxUrl}
+            />
+
+            <br />
+            <AmountRow>
                 {anonymity && <UtxoAnonymity anonymity={anonymity} />}
-            </IOGridCell>
-            <IOGridCell isAccountOwned={isAccountOwned}>
                 {value && (
-                    <FormattedCryptoAmount
+                    <StyledFormattedCryptoAmount
                         value={formatNetworkAmount(value, symbol)}
                         symbol={symbol}
                     />
                 )}
-            </IOGridCell>
-        </>
+            </AmountRow>
+        </GridItem>
     );
 };
 
-type IOSectionProps = {
-    tx: WalletAccountTransaction;
-    inputs: EnhancedVinVout[];
-    outputs: EnhancedVinVout[];
-};
+interface GridGroupWrapperProps {
+    heading?: React.ReactNode;
+    inputsLength?: number;
+    outputsLength?: number;
+    children: ReactElement | ReactElement[];
+}
 
-const IOSection = ({ tx, inputs, outputs }: IOSectionProps) => {
+const IOGridGroupWrapper = ({
+    heading,
+    inputsLength,
+    outputsLength,
+    children,
+}: GridGroupWrapperProps) => (
+    <GridGroup>
+        {heading ? <GridGroupHeading>{heading}:</GridGroupHeading> : null}
+        {inputsLength !== undefined && outputsLength !== undefined ? (
+            <RowGrid>
+                <IOGridItem>
+                    <Translation id="TR_INPUTS" />
+                    {inputsLength >= 0 ? ` • ${inputsLength}` : null}
+                </IOGridItem>
+                <IOGridItem />
+                <IOGridItem>
+                    <Translation id="TR_OUTPUTS" />
+                    {outputsLength >= 0 ? ` • ${outputsLength}` : null}
+                </IOGridItem>
+            </RowGrid>
+        ) : null}
+        {children}
+    </GridGroup>
+);
+
+interface GridRowGroupComponentProps {
+    from?: string;
+    to?: string;
+    symbol: string;
+    amount?: string | ReactNode;
+}
+
+const GridRowGroupComponent = ({ from, to, symbol, amount }: GridRowGroupComponentProps) => {
     const theme = useTheme();
+    const explorerTxUrl = useExplorerTxUrl();
 
-    const { selectedAccount } = useSelector(state => state.wallet);
+    return (
+        <RowGrid>
+            <RowGridItem>
+                <IOAddress txAddress={from} explorerUrl={explorerTxUrl} />
+                <br />
+                {typeof amount === 'string' ? (
+                    <StyledFormattedCryptoAmount value={amount} symbol={symbol} />
+                ) : (
+                    amount
+                )}
+            </RowGridItem>
+            <RowGridItem>
+                <Icon icon="ARROW_RIGHT" size={17} color={theme.TYPE_LIGHT_GREY} />
+            </RowGridItem>
```

Comment:
To center the icon?

```suggestion
            <IconWrapper>
                <Icon icon="ARROW_RIGHT" size={17} color={theme.TYPE_LIGHT_GREY} />
            </IconWrapper>
```

Reply:
It looks weird imho for ETH. It looks good for BTC

Reply:
Done, squashed